consumers/google.go

```diff
@@ -197,27 +199,59 @@ func (d *googleDNSConsumer) applyChange(change *dns.Change) error {
 		return nil
 	}
 
-	_, err := d.client.Changes.Create(params.project, params.zone, change).Do()
-	if err != nil {
-		if strings.Contains(err.Error(), "alreadyExists") {
-			log.Warnf("Cannot update some DNS records (already exist)")
-			return nil
+	additions := make(map[string][]*dns.ResourceRecordSet)
+	deletions := make(map[string][]*dns.ResourceRecordSet)
+
+	for _, add := range change.Additions {
+		hostedZone := d.hostedZoneFor(add.Name)
+		additions[hostedZone] = append(additions[hostedZone], add)
+	}
+	for _, del := range change.Deletions {
+		hostedZone := d.hostedZoneFor(del.Name)
+		deletions[hostedZone] = append(deletions[hostedZone], del)
+	}
+
+	changes := make(map[string]*dns.Change)
+	for _, z := range d.zones {
+		if len(additions[z.Name]) == 0 && len(deletions[z.Name]) == 0 {
+			continue
+		}
+
+		changes[z.Name] = &dns.Change{
+			Additions: additions[z.Name],
+			Deletions: deletions[z.Name],
+		}
+	}
+
+	for z, c := range changes {
+		_, err := d.client.Changes.Create(params.project, z, c).Do()
+		if err != nil {
+			if strings.Contains(err.Error(), "alreadyExists") {
+				log.Warnf("Cannot update some DNS records (already exist)")
+				return nil
+			}
+			return fmt.Errorf("Unable to create change for %s/%s: %v", params.project, z, err)
```

Comment:
yes, will definitely change that. But I would leave out the parallel stuff for now if you don't mind. Is it just for performance or do you think of sth. else?